Hippo.Web/ClientApp/src/components/Order/ChartStringValidation.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { authenticatedFetch } from "../../util/api";
+import { ChartStringValidationModel } from "../../types";
+import { faCircleNotch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+interface ChartStringValidationProps {
+  chartString: string;
+}
+
+type ChartStringStatus = "loading" | "error" | "success" | undefined;
+
+const ChartStringValidation: React.FC<ChartStringValidationProps> = ({
+  chartString,
+}) => {
+  const [chartStringValidation, setChartStringValidation] =
+    React.useState<ChartStringValidationModel>(null);
+  const [status, setStatus] = React.useState<ChartStringStatus>(undefined);
+
+  React.useEffect(() => {
+    if (!chartString) {
+      setStatus(undefined);
+      return;
+    }
+
+    const validateChartString = async () => {
+      setStatus("loading");
+      try {
+        let response = await authenticatedFetch(
+          `/api/order/validateChartString/${chartString}/Debit`,
+          {
+            method: "GET",
+          },
+        );
+        if (response.ok) {
+          const result: ChartStringValidationModel = await response.json();
+          setChartStringValidation(result);
+          setStatus("success");
+        } else {
+          setStatus("error");
+        }
+      } catch (error) {
+        setStatus("error");
+      }
+    };
+
+    validateChartString();
+  }, [chartString]);
+
+  // default to showing loading state on mount
+  if (status === "loading" || status === undefined) {
```

Comment:
if you want to default to loading you can just have the initial `useState("loading")` and get rid of undefined entirely